### ISSUE TITLE
redeploys of the app layer

### ISSUE THREAD

**Issue by ausfestivus** [hashicorp/terraform-azurerm-terraform-enterprise#68]
Per the Azure RA, I have an persistent data layer and the TFE app layer.
If I destroy the app layer and redeploy it via our pipeline, the new deployment doesn't start.

The atlas and archivist service pods are all reporting:

```
Error writing data to auth/kubernetes/login: Error making API request.

URL: PUT http://vault:8200/v1/auth/kubernetes/login
Code: 503. Errors:

* Vault is sealed
! Unable to retrieve token, waiting and trying again
```

Im assuming that the vault couldnt unseal because its a new app layer.
Any ideas on how to recover from this or prevent it from happening in subsequent deploys of the app layer?

I have opened a case with support but figured it wouldnt hurt to ask here either.

**Comment by ausfestivus** [comment]
So, an update on this after a Hashicorp support engagement.

Turns out this problem is occurring because of a problem with the handling of the `encryption_password` variable.

A fix was made in https://github.com/hashicorp/terraform-azurerm-terraform-enterprise/commit/f67e750a0004de6aa359c7e85f63baf9180a342c#diff-9f5ca61a12a3237022b873fc92de3143 and it works thanks to @straubt1 .

I have forked the repo and backported the fix onto v0.1.1 as v0.1.1b https://github.com/Diaxion/terraform-azurerm-terraform-enterprise/releases/tag/v0.1.1b

Can I ask for the same to be done here in the upstream? Our customer wont go into prod with code that isn't from a Hashicorp repo.

**Comment by bnferguson** [comment]
@ausfestivus Thanks for the heads up on this! Got a backport merged and just pushed a new `v0.1.x` release for the Terraform v0.11 line of this module. https://github.com/hashicorp/terraform-azurerm-terraform-enterprise/releases/tag/v0.1.2

Going to go ahead and close this now if that's alright and I hope that gets things going for your customer engagement!
